bindings/java/mongocrypt/build.gradle.kts

```diff
@@ -0,0 +1,337 @@
+                | Version mismatch:
+                | =================
+                |
+                | $version != $gitVersion
+                |
+                | The project version does not match the git tag.
+                |""".trimMargin()
+            throw GradleException(cause)
+        } else {
+            println("Publishing: ${project.name} : $gitVersion")
+        }
+    }
+
+    if (gitVersion == version) {
+        dependsOn(tasks.withType<PublishToMavenRepository>())
+    }
+}
+
+/*
+For security we allow the signing-related project properties to be passed in as environment variables, which
+Gradle enables if they are prefixed with "ORG_GRADLE_PROJECT_".  But since environment variables can not contain
+the '.' character and the signing-related properties contain '.', here we map signing-related project properties with '_'
+to ones with '.' that are expected by the signing plugin.
+*/
+gradle.taskGraph.whenReady {
+    if (allTasks.any { it is Sign }) {
+        val signing_keyId: String? by project
+        val signing_secretKeyRingFile: String? by project
+        val signing_password: String? by project
+
+        allprojects {
+            signing_keyId?.let { extra["signing.keyId"] = it }
+            signing_secretKeyRingFile?.let { extra["signing.secretKeyRingFile"] = it }
+            signing_password?.let { extra["signing.password"] = it }
+        }
+    }
+}
+
+/*
+ * Jna copy or download resources
+ */
+val jnaLibsPath: String = System.getProperty("jnaLibsPath", "")
+val jnaLibsCheck: Boolean = System.getProperties().containsKey("jnaLibsCheck")
+val jnaResources: String = System.getProperty("jna.libary.path", jnaLibsPath)
+val jnaResourcesBuildDir = "$buildDir/jnaLibs/"
+
+// Copy resources to jnaResourcesBuildDir
+val copyResources by tasks.register<Copy>("copyResources") {
+    destinationDir = file(jnaResourcesBuildDir)
+    if (jnaResources.isNotEmpty()) {
+        from(jnaResources)
+        include("**/libmongocrypt.so", "**/libmongocrypt.dylib", "**/mongocrypt.dll")
+    } else {
+        val jnaMapping = mapOf(
+                "libmongocrypt.so" to "linux-x86-64",
+                "mongocrypt.dll" to "win32-x86-64",
+                "libmongocrypt.dylib" to "darwin")
+
+        val copySpecs = jnaMapping.mapTo(mutableListOf(), { copySpec {
+            from("../../../cmake-build")
+            include(it.key)
+            into(it.value)
+        }}).toTypedArray()
+        with(*copySpecs)
+    }
+}
+
+// Download jnaLibs that match the git to jnaResourcesBuildDir
+val downloadJnaLibs by tasks.register<DefaultTask>("downloadJnaLibs")
+downloadJnaLibs.apply {
+
+    val revision = System.getProperty("gitRevision", if (gitVersion == version) gitVersion else gitHash)
+
+    // TODO - Any more libraries? Check the `{OS}-{ARCH}` is correct
+    val jnaMapping = mapOf(
+            "https://s3.amazonaws.com/mciuploads/libmongocrypt/ubuntu1804-64/master/$revision/libmongocrypt.tar.gz" to "linux-x86-64",
```

Comment:
Let's take the oldest platform we can find, so that the glibc dependency is the oldest we will encounter.  Would that be RHEL 7 now?

Reply:
Done - however the rhel 7 version fails on my Ubuntu 18.04 due to:

`java.lang.UnsatisfiedLinkError: libssl.so.10: cannot open shared object file: No such file or directory`